sessionctx/variable/session.go

```diff
@@ -699,7 +699,7 @@ func (s *SessionVars) SetSystemVar(name string, val string) error {
 	case TiDBOptCorrelationThreshold:
 		s.CorrelationThreshold = tidbOptFloat64(val, DefOptCorrelationThreshold)
 	case TiDBOptCorrelationExpFactor:
-		s.CorrelationExpFactor = tidbOptPositiveInt32(val, DefOptCorrelationExpFactor)
+		s.CorrelationExpFactor = int(tidbOptInt64(val, DefOptCorrelationExpFactor))
```

Comment:
Why change it?

Reply:
According to `tidbOptPositiveInt32`, if default value is 1, we cannot set the value to zero successfully.

Reply:
According to `tidbOptPositiveInt32`, if default value is 1, we cannot set the value to zero successfully.